File: Лабораторная 2/WindowsFormsApp5/Port.cs
using System;
using System.Collections.Generic;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace WindowsFormsApp5
{
    class Port
    {
        ClassArray<ITransport> port;
        List<ClassArray<ITransport>> portStages;
        int countPlaces = 20;
        int placeSizeWidth = 210;
        int placeSizeHeight = 80;
        int currentLevel;
        public int getCurrentLevel { get { return currentLevel; } }
        public Port(int countStages)
        {
            portStages = new List<ClassArray<ITransport>>(countPlaces);
            for (int i = 0; i < countStages; i++)
            {
                port = new ClassArray<ITransport>(countPlaces, null);
                portStages.Add(port);
            }
        }
        public void LevelUp()
        {
            if (currentLevel + 1 < portStages.Count)
            {
                currentLevel++;
            }
        }
        public void LevelDown()
        {
            if (currentLevel > 0)
            {
                currentLevel--;
            }
        }
        public int PutShipInPort(ITransport ship)
        {
            return portStages[currentLevel] + ship;
        }
        public ITransport GetShipInPort(int ticket)
        {
            return portStages[currentLevel] - ticket;
        }
        public void Draw(Graphics g, int weidth, int height)
        {
            DrawMarking(g);
            for (int i = 0; i < countPlaces; i++)
            {
                var ship = portStages[currentLevel][i];
                if (ship != null)
                {
                    ship.setPosition(5 + i / 5 * placeSizeWidth + 5, i % 5 * placeSizeHeight + 15);
                    ship.drawShip(g);
                }

            }
        }
        private void DrawMarking(Graphics g)
        {
            Pen pen = new Pen(Color.Black, 3);
            g.DrawString("L" + (currentLevel + 1), new Font("Arial", 30), new SolidBrush(Color.Black), (countPlaces / 5) * placeSizeWidth * 70, 420);
            g.DrawRectangle(pen, 0, 0, (countPlaces / 5) * placeSizeWidth, 480);
            for (int i = 0; i < countPlaces / 5; i++)
            {
                for (int j = 0; j < 6; ++j)
                {
                    g.DrawLine(pen, i * placeSizeWidth, j * placeSizeHeight, i * placeSizeWidth + 110, j * placeSizeHeight);
                    if (i < 5)
                    {
                        g.DrawString((i * 5 + j + 1).ToString(), new Font("Arial", 30), new SolidBrush(Color.Black), i * placeSizeWidth + 30, j * placeSizeHeight + 20);
                    }
                }
                g.DrawLine(pen, i * placeSizeWidth, 0, i * placeSizeWidth, 400);
            }

        }
    }

}

File: lab_2_TP/src/lab_2_TP/Port.java
package lab_2_TP;

import java.awt.Color;
import java.awt.Graphics;
import java.io.BufferedInputStream;
import java.io.FileInputStream;
import java.io.FileNotFoundException;
import java.io.FileOutputStream;
import java.io.IOException;
import java.io.ObjectInputStream;
import java.io.ObjectOutputStream;
import java.io.Serializable;
import java.util.ArrayList;

public class Port implements Serializable{
	
	 	 
	 	int countPlaces = 10;
	 	ArrayList<ClassArray<ITransport>> portStages;
	 	int placeSizeWidth = 210;
	 	int placeSizeHeight = 80;
	 	int currentLevel; 
	 	public Port(int countStages)
	 	{
	 			 	portStages = new ArrayList<ClassArray<ITransport>>(countStages);
	 		  		for (int i = 0; i < countStages; i++)
	 		  		{
	 		  			portStages.add(new ClassArray<ITransport>(countPlaces, null));
	 		  		}
	 	}
	 	public int getCurrentLevel(){
	 		 			return currentLevel;	
	 		  	}
	 		  
	 		  	public void levelUp()
	 		  	{
	 		  		if (currentLevel + 1 < portStages.size())
	 		  			{
	 		  			currentLevel++;
	 		  			}
	 		  	}
	 		  
	 		  	public void levelDown()
	 		  	{
	 		  		if (currentLevel > 0)
	 		  			{
	 		  			currentLevel--;			
	 		  			}
	 		   	}
	 		  	public int PutLinerInPort(ITransport ship) throws PortOverflowException
	 	 	{
	 	 		return portStages.get(currentLevel).plus(portStages.get(currentLevel), ship);
	 	 	}
	 	 
	 		  	public ITransport GetLinerInPort(int ind) throws PortIndexOutOfRangeException
	 	 	{
	 	 		return portStages.get(currentLevel).minus(portStages.get(currentLevel), ind);
	 	 	}
	 	 
	 	 	public void Draw(Graphics g,int width,int height)
	 	 	{
	 	 		DrawMarking(g);
	 	 		for(int i = 0; i < countPlaces; i++)
	 	 		{
	 	 			ITransport ship = portStages.get(currentLevel).getLiner(i);
	 	 			if (ship != null)
	 	 			{
	 	 				ship.setPosition(0 + i / 5 * placeSizeWidth + 8, i % 5 * placeSizeHeight + 7);
	 	 				ship.drawShip(g);
	 	 			}
	 	 		}
	 	 		
	 	 	}	 
	 	 	public void DrawMarking(Graphics g)
	 	 	{
	 	 		g.setColor(Color.BLACK);
	 	 		g.drawRect( 0, 0, (countPlaces / 5) * placeSizeWidth, 480);
	 	 		for(int i = 0; i < countPlaces / 5; i++)
	 	 		{
	 	 			for(int j = 0; j < 6; j++)
	 	 			{
	 	 				g.drawLine( i * placeSizeWidth,j* placeSizeHeight,i*placeSizeWidth+110,j*placeSizeHeight);
	 	 			}
	 	 			g.drawLine( i * placeSizeWidth, 0, i * placeSizeWidth,400);
	 	 		}
	 	 		
		 	}
	 	 	public boolean save(String fileName) throws IOException {
	 	 		 
	 	 		 			FileOutputStream save = null;
	 	 		 			try {
	 	 		 				save = new FileOutputStream(fileName);
	 	 		 			} catch (FileNotFoundException e) {
	 	 		 				// TODO Auto-generated catch block
	 	 		 				e.printStackTrace();
	 	 		 			}
	 	 		 			ObjectOutputStream obSave = new ObjectOutputStream(save);
	 	 		 			System.out.println(portStages.get(0).getLiner(0).getInfo());
	 	 		 			obSave.writeObject(portStages);
	 	 		 
	 	 		 			return true;
	 	 		 		}
	 	 		 
	 	 		 		public boolean load(String filename) {
	 	 		 			try {
	 	 		 				ObjectInputStream obLoad = new ObjectInputStream(new BufferedInputStream(new FileInputStream(filename)));
	 	 		 				try {
	 	 		 					portStages = (ArrayList<ClassArray<ITransport>>)obLoad.readObject();
	 	 		 					System.out.println(portStages.get(0).getLiner(0).getInfo());
	 	 		 				} catch (ClassNotFoundException e) {
	 	 		 					// TODO Auto-generated catch block
	 	 		 					e.printStackTrace();
	 	 		 				}
	 	 		 			} catch (IOException e) {
	 	 		 				// TODO Auto-generated catch block
	 	 		 				e.printStackTrace();
	 	 					}
	 	 		 			
	 	 		 			return true;
	 	 		 		}
	 	 
	 	 
}